### PR TITLE
Expose max nesting depth in hash function to plugin

### PR DESCRIPTION
**Pull request by ustcfy**
Closes https://github.com/NVIDIA/spark-rapids-jni/issues/2681

This PR includes the following changes:

1. Moved the definition of `MAX_NESTED_DEPTH` from the specific hash function to `hash.hpp`.
2. Exposed the value of this parameter to plugin.

### CODE REVIEW

**Comment by firestarman** [comment]
build

**Review by firestarman — DISMISSED** [review]
LGTM

File: src/main/cpp/src/HashJni.cpp [hunk 1]
```diff
@@ -21,6 +21,15 @@
 
 extern "C" {
 
+JNIEXPORT jint JNICALL Java_com_nvidia_spark_rapids_jni_Hash_getMaxNestedDepth(JNIEnv* env, jclass)
+{
+  try {
+    cudf::jni::auto_set_device(env);
```
Comment (res-life) [inline src/main/cpp/src/HashJni.cpp:27]: This feature has nothing to do with GPU(device). 
```suggestion
```

File: src/main/cpp/src/HashJni.cpp [hunk 2]
```diff
@@ -21,6 +21,14 @@
 
 extern "C" {
 
+JNIEXPORT jint JNICALL Java_com_nvidia_spark_rapids_jni_Hash_getMaxNestedDepth(JNIEnv* env, jclass)
+{
+  try {
+    return spark_rapids_jni::MAX_NESTED_DEPTH;
+  }
+  CATCH_STD(env, 0);
```
Comment (ttnghia) [inline src/main/cpp/src/HashJni.cpp:29]: This is a trivial statement, nothing to try/catch.
```suggestion
    return spark_rapids_jni::MAX_NESTED_DEPTH;
```

**Comment by ustcfy** [comment]
build

**Review by ttnghia — APPROVED** [review]

**Comment by pxLi** [comment]
build

**Review by res-life — APPROVED** [review]

**Pull request merged**